dotnet/targets/Xamarin.Shared.Sdk.DefaultItems.targets

```diff
@@ -23,6 +23,20 @@
     <None Include="@(ObjcBindingCoreSource)" />
   </ItemGroup>
 
+  <PropertyGroup Condition="'$(_PlatformName)' == 'MacCatalyst'">
+    <EnableDefaultMacCatalystReleaseEntitlements Condition="'$(EnableDefaultMacCatalystReleaseEntitlements)' == ''">True</EnableDefaultMacCatalystReleaseEntitlements>
+    <EnableDefaultMacCatalystDebugEntitlements Condition="'$(EnableDefaultMacCatalystDebugEntitlements)' == ''">True</EnableDefaultMacCatalystDebugEntitlements>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(EnableDefaultMacCatalystDebugEntitlements)' == 'True' and '$(Configuration)' == 'Debug'">
+    <CustomEntitlements Include="com.apple.security.get-task-allow" Type="boolean" Value="true" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(EnableDefaultMacCatalystReleaseEntitlements)' == 'True' and '$(Configuration)' == 'Release'">
+    <CustomEntitlements Include="com.apple.security.app-sandbox" Type="boolean" Value="true" />
+    <CustomEntitlements Include="com.apple.security.network.client" Type="boolean" Value="true" />
```

Comment:
I have mix feelings regarding this one: https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_security_network_client

As per the docs:

```
For TCP sockets, the com.apple.security.network.client and [com.apple.security.network.server](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_security_network_server) entitlements restrict only the initiation of a network connection
```

Afaik it is not needed for network requests but to init the tcp socket connection. If this is for the webassembly scenario it makes sense, if it is for network connections, it does not @dustin-wojciechowski  can you clarify?

Reply:
This entitlement was specifically required for Maui Blazor applications as noted here: https://github.com/dotnet/maui/issues/13094. User was not able to proceed past splash screen without it in Release config.

Reply:
That is exactly my concern, opening all applications to init tcp connections to a local host because blazer needs it is problematic. An apple review could complain about it stating that is not needed in an app that does not have blazor.